### PR TITLE
Add fastboot support

### PR DESCRIPTION
**Pull request by ryanto**
Adds fastboot support by creating an adapter cache and then using it to instantly fulfill client side xhr requests.

### CODE REVIEW

File: fastboot-tests/adapter-test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      .textContent;
+
+    let cache = JSON.parse(shoebox);
+    let keys = Object.keys(cache.queries);
+
+    assert.equal(keys.length, 1);
+    assert.ok(cache.queries['post::filter[popular]=true']);
+  });
+
+});
```
Comment (ryanto) [inline fastboot-tests/adapter-test.js]: I got this test idea from liquid-fire. I think it's a pretty good way to test fastboot apps. 

https://github.com/ember-animation/liquid-fire/blob/master/fastboot-tests/demos-test.js

**Comment by ryanto** [comment]
ok, this is ready for review!

File: tests/dummy/config/environment.js [hunk 2]
```diff
@@ -10,7 +10,8 @@ module.exports = function(environment) {
     locationType: 'router-scroll',
     historySupportMiddleware: true,
     routerScroll: {
-      scrollElement: '#docs-viewer__scroll-body'
+      //scrollElement: '#docs-viewer__scroll-body'
+      scrollElement: 'window'
```
Comment (ryanto) [inline tests/dummy/config/environment.js]: There's a bug in router-scroll where it doesn't work with scrollElement+fastboot. I'm going to fix that, but right now this is the best solution.

**Comment by samselikoff** [comment]
Add findAll to fastboot support

**Comment by ryanto** [comment]
fastboot adapter now supports all queries

**Pull request merged**